docs/GLOSSARY.md

```diff
@@ -1,76 +1,88 @@
 # Glossary
 
-Index of terminology used in Randovania. Certain terms sometimes are used only in the GUI, with more technical ones being used in code or CLI.
+Index of terminology used in Randovania. Some terms are used only in the GUI, with more technical ones being used in code or CLI.
 
 ## Area
 
-TODO
+A small chunk of a game's playable space, often only a single room. May contain one or more [Nodes](#node).
+
+Examples:
+- Phazon Mining Tunnel in Metroid Prime
+- Big Pink in Super Metroid
 
 ## Configuration
 
-TODO
+Has two meanings:
+
+- Colloquially, a term referring to the many types of settings a player may use when generating a game. This includes trick settings, item pool settings, starting items and location, [Gameplay and Cosmetic Patches](#game-patches), and many other things.
+
+- In code, the Configuration class that each game must provide, derived from the BaseConfiguration class. This is where game-specific parts of the [Preset](#preset) are defined.
 
 ## Dangerous Action
 
-TODO
+Any action that the [Generator](#generator) might take which could potentially leave it stranded; a one-way action. See also [Point of No Return](#point-of-no-return).
 
 ## Database
 
-TODO
+The collection of data that represents a game's [Logic](#logic), including the layout of [Worlds](#world) and [Areas](#area), the [Resources](#resource) a player has to progress, the locations of [Pickups](#pickup) and spawn points, and other information.
 
 ## Event
 
-TODO
+A logical [Resource](#resource) representing an event that may be triggered in game. Events can only be completed once, and cannot be undone.
```

Comment:
> A Resource ...
logical has no meaning here.

Also, you're defining event using the word event. What is an in-game event?

Reply:
You're right, that was a bit of a circular definition. Reworded and clarified.